ldap/client/api/src/main/java/org/apache/directory/ldap/client/api/LdapNetworkConnection.java

```diff
@@ -516,31 +522,59 @@ private static LdapConnectionConfig buildConfig( String server, int port, boolea
     }
 
 
+    //-------------------------- The methods ---------------------------//
     /**
      * Create the connector
      */
-    private void createConnector() throws LdapException
+    private IoConnector createConnector() throws LdapException
     {
         // Use only one thread inside the connector
-        connector = new NioSocketConnector( 1 );
+        IoConnector conn = new NioSocketConnector( 1 );
```

Comment:
Yep. There might be many concurrent connectors at the same time, until we find a workable connection. When that happens, we set the class variable to the connector we are going to use and then we drop (dispose) all the other connectors.

So it's to get around a potential race condition.